src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -48,6 +48,11 @@ var ReactDOMTextComponent = function(text) {
   this._mountIndex = 0;
   this._closingComment = null;
   this._commentNodes = null;
+
+  if (__DEV__) {
+    // validateDOMNesting uses this:
+    this._isWhitespace = typeof text === 'string' && text.trim().length === 0;
```

Comment:
Just curious, since this might be somewhat of a hot path. Have you benchmarked this against `REGEX.test(text)`? With REGEX being a constant and not inline.